src/heartbeats.cpp

```diff
@@ -0,0 +1,780 @@
+    : mClient(client)
+{
+}
+#ifdef ENABLE_SYNC
+void MegaBackupMonitor::onSyncBackupRegistered(int syncTag, handle backupId)
+{
+    bool needsAdding = true;
+
+    // remove the tag from the set of in-flight sync registrations
+    mPendingSyncPuts.erase(syncTag);
+
+    if (ISUNDEF(backupId))
+    {
+        LOG_warn << "Received invalid id for sync with tag: " << syncTag;
+        needsAdding = false;
+    }
+
+    if (!ISUNDEF(backupId) && mHeartBeatedSyncs.find(syncTag) != mHeartBeatedSyncs.end())
+    {
+        mHeartBeatedSyncs[syncTag]->setBackupId(backupId);
+        needsAdding = false;
+    }
+
+    if (needsAdding)
+    {
+        for (const auto &hBPair : mHeartBeatedSyncs)
+        {
+            if (hBPair.second->backupId() == backupId)
+            {
+                needsAdding = false;
+                break;
+            }
+        }
+
+        if (needsAdding)
+        {
+            //create new HeartBeatSyncInfo
+            mHeartBeatedSyncs[syncTag] = std::make_shared<HeartBeatSyncInfo>(syncTag, backupId);
+        }
+    }
+
+    // store the id in the sync configuration
+    if (!ISUNDEF(backupId))
+    {
+        mClient->updateSyncBackupId(syncTag, backupId);
+    }
+
+    // handle pending updates
+    const auto &pendingSyncPair = mPendingSyncUpdates.find(syncTag);
+    if (pendingSyncPair != mPendingSyncUpdates.end())
+    {
+        if (!ISUNDEF(backupId))
+        {
+            MegaBackupInfo &info = *pendingSyncPair->second.get();
+            info.setBackupId(backupId);
+            updateBackupInfo(info);
+        }
+        else
+        {
+            LOG_warn << "discarding heartbeat update for pending sync: no valid id received for sync: " << pendingSyncPair->second->localFolder();
+
+        }
+        mPendingSyncUpdates.erase(pendingSyncPair);
+    }
+}
+#endif
+
+void MegaBackupMonitor::digestPutResult(handle backupId)
+{
+    // get the tag from queue of pending puts
+    assert(mPendingBackupPutCallbacks.size());
+    auto putResultCallback = mPendingBackupPutCallbacks.front();
+    mPendingBackupPutCallbacks.pop_front();
+
+
+    //call corresponding callback
+    putResultCallback(backupId);
+}
+
+void MegaBackupMonitor::updateBackupInfo(const MegaBackupInfo &info)
+{
+    string localFolderEncrypted(mClient->cypherTLVTextWithMasterKey("lf", info.localFolder()) );
+    string deviceIDEncrypted(mClient->cypherTLVTextWithMasterKey("de", mClient->getDeviceid()) );
+    string nameEncrypted(mClient->cypherTLVTextWithMasterKey("na", info.name()) );
+
+    mClient->reqs.add(new CommandBackupPut(mClient, info.backupId(), info.type(), info.megaHandle(),
+                                           localFolderEncrypted.c_str(),
+                                           deviceIDEncrypted.c_str(), nameEncrypted.c_str(),
+                                           info.state(), info.subState(), info.extra().c_str()
+                                           ));
```

Comment:
DO NOT MERGE: only changed values need to be updated. Otherwise, for instance, if a backup is renamed in the backup center, it'll be overridden here

Reply:
I think this comment is obsolete, isn't it?

Reply:
So I think. Name is no longer send through this